metadata-ingestion/src/datahub/ingestion/api/registry.py

```diff
@@ -30,6 +32,19 @@ def is_enabled(self, key: str) -> bool:
         tp = self._mapping[key]
         return not isinstance(tp, Exception)
 
+    def load(self, entry_point_key: str) -> None:
+        for entry_point in pkg_resources.iter_entry_points(entry_point_key):
+            name = entry_point.name
+            plugin_class = None
+
+            try:
+                plugin_class = entry_point.load()
```

Comment:
I removed the subclass and abstract class checks because we're being explicit about the class, but if you want me to add them back in, I can.

The subclass check was a little annoying because we have to pass in the type, unless there's a way to resolve the generic at runtime.

Reply:
Yep that should be fine